src/marqo/tensor_search/tensor_search.py

```diff
@@ -1711,11 +1771,73 @@ def create_empty_query_response(queries: List[BulkSearchQueryEntity]) -> List[Di
         )
     )
 
+
+def add_prefix_to_queries(config: Config, queries: List[BulkSearchQueryEntity]) -> List[BulkSearchQueryEntity]:
+    """
+    Makes a new list of queries
+    Adds prefix to the q of all BulkSearchQueryEntity objects in the list.
+    Determines prefix with defaults from index_info textQueryPrefix (per query).
+    Only adds prefix to text queries (not images)
+    """
+    prefixed_queries = []
+    for q in queries:
+        # Determine what prefix to use
+        index_info = get_index_info(config=config, index_name=q.index)
+        text_query_prefix = determine_text_query_prefix(q.textQueryPrefix, index_info)
+        if text_query_prefix is None:
+            text_query_prefix = ""
+        
+        treat_urls_as_images = index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images]
+        # Add prefix to q if applicable
+        if q.q is None:
+            prefixed_q = q.q
+        elif isinstance(q.q, str):
+            if treat_urls_as_images and _is_image(q.q):
+                # Images get no prefix
+                prefixed_q = q.q
+            else:
+                # Single text query: add prefix
+                prefixed_q = f"{text_query_prefix}{q.q}"
+        else:  # is dict:
```

Comment:
Looks like dictionary is a variation of a single string query. You could extract the prefixing logic for one string into another function, and call that for each dict entry (or just once if query is one string) to avoid implementing the same logic twice

Reply:
Not sure if this would be worth it, as:
1. The `treat_urls_as_images` check would happen for every single dict entry, which is inefficient, as opposed to just doing it once at the start, which I'm doing now.
2. Removing that, the only real logic is checking if `_is_image(q)`, which I don't think is enough to make a function for.